config/telemetry.go

```diff
@@ -0,0 +1,405 @@
+		MetricsPrefix:                      c.MetricsPrefix,
+		StatsdAddr:                         c.StatsdAddr,
+		StatsiteAddr:                       c.StatsiteAddr,
+		PrometheusPort:                     c.PrometheusPort,
+		PrometheusRetentionTime:            c.PrometheusRetentionTime,
+	}
+}
+
+// Merge combines all values in this configuration with the values in the other
+// configuration, with values in the other configuration taking precedence.
+// Maps and slices are merged, most other values are overwritten.
+func (c *TelemetryConfig) Merge(o *TelemetryConfig) *TelemetryConfig {
+	if c == nil {
+		if o == nil {
+			return nil
+		}
+		return o.Copy()
+	}
+	if o == nil {
+		return c.Copy()
+	}
+
+	r := c.Copy()
+
+	r.Disable = o.Disable
+
+	if o.CirconusAPIApp != "" {
+		r.CirconusAPIApp = o.CirconusAPIApp
+	}
+	if o.CirconusAPIToken != "" {
+		r.CirconusAPIToken = o.CirconusAPIToken
+	}
+	if o.CirconusAPIURL != "" {
+		r.CirconusAPIURL = o.CirconusAPIURL
+	}
+	if o.CirconusBrokerID != "" {
+		r.CirconusBrokerID = o.CirconusBrokerID
+	}
+	if o.CirconusBrokerSelectTag != "" {
+		r.CirconusBrokerSelectTag = o.CirconusBrokerSelectTag
+	}
+	if o.CirconusCheckDisplayName != "" {
+		r.CirconusCheckDisplayName = o.CirconusCheckDisplayName
+	}
+	if o.CirconusCheckForceMetricActivation != "" {
+		r.CirconusCheckForceMetricActivation = o.CirconusCheckForceMetricActivation
+	}
+	if o.CirconusCheckID != "" {
+		r.CirconusCheckID = o.CirconusCheckID
+	}
+	if o.CirconusCheckInstanceID != "" {
+		r.CirconusCheckInstanceID = o.CirconusCheckInstanceID
+	}
+	if o.CirconusCheckSearchTag != "" {
+		r.CirconusCheckSearchTag = o.CirconusCheckSearchTag
+	}
+	if o.CirconusCheckTags != "" {
+		r.CirconusCheckTags = o.CirconusCheckTags
+	}
+	if o.CirconusSubmissionInterval != "" {
+		r.CirconusSubmissionInterval = o.CirconusSubmissionInterval
+	}
+	if o.CirconusSubmissionURL != "" {
+		r.CirconusSubmissionURL = o.CirconusSubmissionURL
+	}
+	r.DisableHostname = o.DisableHostname
+	if o.DogstatsdAddr != "" {
+		r.DogstatsdAddr = o.DogstatsdAddr
+	}
+	if len(o.DogstatsdTags) != 0 {
+		r.DogstatsdTags = o.DogstatsdTags
+	}
+	r.FilterDefault = o.FilterDefault
+	if len(o.AllowedPrefixes) != 0 {
+		r.AllowedPrefixes = o.AllowedPrefixes
+	}
+	if len(o.BlockedPrefixes) != 0 {
+		r.BlockedPrefixes = o.BlockedPrefixes
+	}
+	if o.MetricsPrefix != "" {
+		r.MetricsPrefix = o.MetricsPrefix
+	}
+	if o.StatsdAddr != "" {
+		r.StatsdAddr = o.StatsdAddr
+	}
+	if o.StatsiteAddr != "" {
+		r.StatsiteAddr = o.StatsiteAddr
+	}
+
+	if o.PrometheusRetentionTime.Nanoseconds() > 0 {
+		r.PrometheusRetentionTime = o.PrometheusRetentionTime
+	}
+	if o.PrometheusPort != 0 {
+		r.PrometheusPort = o.PrometheusPort
+	}
+
+	return r
+}
+
+func (c *TelemetryConfig) GoString() string {
+	if c == nil {
+		return "(*TelemetryConfig)(nil)"
+	}
+	return fmt.Sprintf("&TelemetryConfig{"+
+		"Disable:%v, "+
+		"CirconusAPIApp:%s, "+
+		"CirconusAPIToken:%s, "+
```

Comment:
are we worried about leaking the API token in logs here?

Reply:
Thanks for spotting this issue. The output value has been replaced by `<empty>` and `<configured>`.